pkg/embeddedcluster/util.go

```diff
@@ -126,6 +163,7 @@ func startClusterUpgrade(ctx context.Context, newcfg embeddedclusterv1beta1.Conf
 			ClusterID:                 current.Spec.ClusterID,
 			MetricsBaseURL:            current.Spec.MetricsBaseURL,
 			AirGap:                    current.Spec.AirGap,
+			Artifacts:                 artifacts,
```

Comment:
Sounds like we will be adding a field to the EC installation object for the metadata override, so this will likely need to be updated to account for that.